data_safe_haven/infrastructure/programs/imperative_shm.py

```diff
@@ -147,7 +147,12 @@ def deploy(self) -> None:
         try:
             graph_api.create_application(
                 self.context.entra_application_name,
-                application_scopes=["Group.ReadWrite.All"],
+                application_scopes=[
+                    "Application.ReadWrite.All",
+                    "AppRoleAssignment.ReadWrite.All",
+                    "Directory.ReadWrite.All",
+                    "Group.ReadWrite.All",
+                ],
```

Comment:
These new scopes are required for the application to do the new azuread stuff?

Reply:
In-line explanatory comments added in bc3db5a52.